pkgs/applications/networking/qv2ray/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, fetchFromGitHub
+, qt5
+, cmake
+, clang
+, grpc
+, protobuf
+, openssl
+, pkgconfig
+, c-ares
+, abseil-cpp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "qv2ray";
+  version = "2.5.0";
+
+  src = fetchFromGitHub {
+    owner = "Qv2ray";
+    repo = "Qv2ray";
+    rev = "v${version}";
+    sha256 = "12x72rsd5iaj06qmma8x6g7syqaiixnfb37f3ppx8x00c77s78va";
+    fetchSubmodules = true;
+  };
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DQV2RAY_DISABLE_AUTO_UPDATE=on"
+    "-DQV2RAY_TRANSLATION_PATH=$(out)/share/qv2ray/lang"
+  ];
+
+  preConfigure = ''
+    export _QV2RAY_BUILD_INFO_="Qv2ray Nixpkgs"
+    export _QV2RAY_BUILD_EXTRA_INFO_="(Nixpkgs build) $(uname -a | cut -d " " -f3,13)"
+  '';
+
+  installPhase = ''
+    cmake --install .
+  '';
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [ 
+    cmake
+    clang
+    qt5.qmake 
+    qt5.qttools
+    protobuf
+    grpc
+    pkgconfig
+    c-ares
+    abseil-cpp
```

Comment:
libraries like abseil-cpp (those which contain code that is embedded in the resulting binaries) are supposed to go to `buildInputs`.
(as opposed to build tools which are executed during the build, like cmake, which you correctly placed in `nativeBuildInputs`. )

If unsure, you can try to build with `strictDeps  = true`, and it should fail to build if you made a mistake.